### PR TITLE
Add tracing for shared locks in `id_lock_map`

### PR DESCRIPTION
**Pull request by jbajic**
## Summary of changes

Extension of issue https://github.com/neondatabase/neon/issues/7108 in tracing to exclusive lock in `id_lock_map` was added, to add the same for shared locks. It was mentioned in the comment https://github.com/neondatabase/neon/pull/7397#discussion_r1587961160

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: storage_controller/src/id_lock_map.rs [hunk 1]
```diff
@@ -8,14 +8,15 @@ use crate::service::RECONCILE_TIMEOUT;
...
+/// A wrapper around `OwnedRwLockWriteGuard` used for tracking the
+/// operation that holds the lock, and print a warning if it exceeds
+/// the LOCK_TIMEOUT_ALERT_THRESHOLD time
+pub struct TracingWriteGuard<T: Display> {
     guard: tokio::sync::OwnedRwLockWriteGuard<Option<T>>,
     start: Instant,
 }
 
-impl<T: Display> WrappedWriteGuard<T> {
+impl<T: Display> TracingWriteGuard<T> {
```
Comment (VladLazar) [inline storage_controller/src/id_lock_map.rs:19]: I would use the `TracingExclusiveGuard` and `TracingSharedGuard` names, but don't feel super strongly.

**Review by VladLazar — COMMENTED** [review]
Thanks again, Jure!

**Review by VladLazar — APPROVED** [review]
(meant to approve)

**Comment by jbajic** [comment]
Just checking if there is some issue with the PR @VladLazar ?

**Comment by VladLazar** [comment]
> Just checking if there is some issue with the PR @VladLazar ?

The issue is that I went on holiday and forgot about it. Let me sanity check it again and merge.

**Comment by VladLazar** [comment]
Merged. Apologies for the delay and thank you, Jure!

**Pull request merged**